Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1220,18 +1222,28 @@ private boolean readAndStripParam(Uri data, Activity activity) {
             // Capture the intent URI and extra for analytics in case started by external intents such as  google app search
             try {
                 if (data != null) {
+                    boolean foundSchemeMatch;
+                    boolean skipThisPath = false;
                     if (externalUriWhiteList_.size() > 0) {
-                        String externalIntentScheme = Uri.parse(data.toString()).getScheme();
-                        if (externalIntentScheme != null && externalUriWhiteList_.contains(externalIntentScheme)) {
-                            sessionReferredLink_ = data.toString();
-                            prefHelper_.setExternalIntentUri(data.toString());
-                        }
+                        foundSchemeMatch = externalUriWhiteList_.contains(data.getScheme());
                     } else {
+                        foundSchemeMatch = true;
+                    }
+                    if (skipExternalUriPaths_.size() > 0) {
+                        for (String path : skipExternalUriPaths_) {
+                            String externalPath = data.getPath();
```

Comment:
Actually this didn't solve the problem. Let's say we have the Uri `sth://callback`. The `data.getPath()` would simply return `/`. I guess `data.getHost` should be used?